File: Algebra_MultiplyFraction.py
#!/usr/bin/python
from MultiplyFraction import TimesFraction

##
 # Multiplying fractions
 ##
numerators   = [16, 20, 27, 20]
denominators = [9, 9, 640, 7]
fractions = {'numerators':numerators, 'denominators':denominators}

print('\n    Solving:\n')
# Print as fraction
for numerator in fractions['numerators']: print('{:13d}'.format(numerator), end='')
print('{}{:>12}'.format('\n', ' '), end='')
for wasted in range(len(numerators)-1): print('{}'.format('-     X      '), end='')
print('{:>1}'.format('-'))
for denominator in fractions['denominators']: print('{:13d}'.format(denominator), end='') 
print('\n\n')

# use the MultiplyFraction class
mul_fract = TimesFraction(fractions)
fraction = mul_fract.doMultiply()

print('{:25d}'.format(fraction['numerator']))
print('{:>25}'.format('Answer   =   -'))
print('{:25d}'.format(fraction['denominator']))


print('\n\n')

File: Algebra_MultiplyFraction.vb
Module Algebra_multiply_fraction

    Sub Main()
        ''
        ' Multiplying fractions
        ''
        Dim numerators = {16, 20, 27, 20}
        Dim denominators = {9, 9, 640, 7}
        Dim fractions As New Dictionary(Of String, Integer())
        fractions.Add("numerators", numerators)
        fractions.Add("denominators", denominators)

        Console.WriteLine("    Solving:")
        ' Print as fraction
        For Each numerator In fractions.item("numerators")
            Console.Write(String.Format("{0,13}", numerator))
        Next
        Console.Write(Environment.NewLine & String.Format("{0,12}", " "))
        For wasted = 0 To numerators.Count - 2
            Console.Write(String.Format("{0}", "-     X      "))
        Next
        Console.WriteLine(String.Format("{0,1}", "-"))
        For Each denominator In fractions.item("denominators")
            Console.Write(String.Format("{0,13}", denominator))
        Next
        Console.WriteLine("")

        ' use the MultiplyFraction Class
        Dim mul_fract As New MultiplyFraction
        mul_fract._init_(fractions)
        Dim fraction = mul_fract.doMultiply()

        Console.WriteLine(Environment.NewLine)

        Console.WriteLine(String.Format("{0,25}", fraction.Item("numerator")))
        Console.WriteLine(String.Format("{0,25}", "Answer   =   -"))
        Console.WriteLine(String.Format("{0,25}", fraction.Item("denominator")))

    End Sub

End Module
